src/language/class-modifiers.md

```diff
@@ -0,0 +1,258 @@
+---
+title: Class modifiers
+description: Modifier keywords for class declarations to control external library access.
+---
+
+Class modifiers control how a class or a mixin can be subtyped, both
+[from within its own library](#abstract), and from outside of the library where
+it's defined.
+
+Modifiers prepend a class or mixin declaration.
+For example, writing `abstract class` defines an abstract class.
+The full set of modifiers that can appear before a class declaration are:
+
+- `abstract`
+- `base`
+- `final`
+- `interface`
+- `sealed`
+- [`mixin`][class, mixin, or mixin class]
+
+Only the `base` modifier can appear before a mixin declaration. The modifiers do
+not apply to other declarations like `enum`, `typedef`, or `extension`.
+
+When deciding whether to use class modifiers, consider the intended uses of the
+class, and what behaviors the class needs to be able to rely on.
+
+## No modifier
+
+Use a `class` or `mixin` declaration without a modifier to allow unrestricted
+permission to create subtype relationships from any library.
+By default, you can:
+
+- [Construct][] new instances of a class.
+- [Extend][] a class to create a new subtype.
+- [Implement][] a class or mixin's interface.
+- [Mix in][mixin] a mixin or mixin class.
+
+## `abstract`
+
+Use the `abstract` modifier to define a class that can’t be constructed from any
+library (it's own, as well as outside libraries).
+Abstract classes are useful for defining interfaces, often with some
+implementation. Abstract classes often have [abstract methods][].
+
+```dart
+// Library a.dart
+abstract class Vehicle { 
+  void moveForward();     // Abstract method.
+}
+```
+
+```dart
+// Library b.dart
+import 'a.dart';
+
+var myCar = Vehicle();     // Error: Cannot be constructed
+
+class Car extends          // Can be extended
+  Vehicle {
+    int passengers;
+    ...
+}
+
+class MockVehicle          // Can be implemented
+  implements Vehicle {
+    @override
+    void moveForward(int meters) { ... }
+}
+```
+
+If you want your abstract class to appear to be instantiable,
+define a [factory constructor][].
+
+## `base`
+
+Use the `base` modifier to enforce inheritance of a class or mixin's implementation.
+The `base` modifier disallows implementation outside of the current library, which
+guarantees:
+
+- The base class constructor is called whenever an instance of a subtype of the
+class is created.
+- That all implemented private members exist in subtypes.
+- Adding a new implemented member to a `base` class will not break subtypes (except
+  when they already declare a member with the same name and an incompatible
+  signature), since all subtypes will inherit the new member.
```

Comment:
Can't really active-voice the first bullet again since it could be anyone, most likely not "you", when we're talking about restricting external libraries. Other notes incorporated though